message_handlers/group/tip.ts

```diff
@@ -164,12 +203,86 @@ export const tip = async (bot: TelegramBot, update: Update) => {
   });
 };
 
+async function buildTransactionConfig(
+  tipperWallet: Wallet,
+  recipientWallet: Wallet,
+  amount: number,
+  tokenContract?: Contract
+) {
+  let data = null;
+  let transactionConfig = null;
+  if (tokenContract) {
+    await transactionService.approve(
+      tokenContract,
+      tipperWallet.address,
+      process.env.CONTRACT_ADDRESS!,
+      amount
+    );
+
+    data = transactionService.tipByToken(
+      recipientWallet.address,
+      tokenContract.options.address,
+      amount
+    );
+  } else {
+    data = transactionService.tipByContract(recipientWallet.address);
```

Comment:
set this as initial value instead of having an `else` block.

`let data = transactionService.tipByContract(recipientWallet.address);`